onnx/defs/nn/defs.cc

```diff
@@ -809,37 +818,44 @@ void convTransposeShapeInference(InferenceContext& ctx) {
   *final_output_shape->add_dim() = input_shape.dim(0);
   *final_output_shape->add_dim() =
       ctx.getInputType(1)->tensor_type().shape().dim(
-          1); // channels should be the second dim of second input.
+          1) * group; // channels should be the second dim of second input multiply group.
 
-  int size_of_output = static_cast<int>(output_shape.size());
-  if (size_of_output > 0) {
+  int size_of_output;
+  if (output_shape_presented) {
+    size_of_output = static_cast<int>(output_shape.size());
     for (int i = 0; i < size_of_output; ++i) {
-      if (output_shape[i] < input_shape.dim(i + 2).dim_value()) {
-        // TODO: throw exception?
-        return; // output shape value cannot be smaller than the input shape
-                // value
+      if (input_shape.dim(i + 2).has_dim_value()) {
+        if (output_shape[i] < input_shape.dim(i + 2).dim_value()) {
+          // TODO: throw exception?
+          return; // output shape value cannot be smaller than the input shape
+                  // value
+        }
       }
-
       final_output_shape->add_dim()->set_dim_value(output_shape[i]);
     }
-    return; // assume no need to proceed further when the output shape is given.
+    return;
   }
-
-  int kernel_shape_size = static_cast<int>(kernel_shape.size());
-  for (int i = 0; i < kernel_shape_size; ++i) {
-    auto newdim = final_output_shape->add_dim();
-    if (!input_shape.dim(2 + i).has_dim_value()) {
-      continue;
+  else
+  {
+    size_of_output = input_shape.dim_size() - 2;
+    for (int i = 0; i < size_of_output; ++i)
+    {
+      if (input_shape.dim(i + 2).has_dim_value()) {
+        int64_t output_shape_dim =
+            strides[i] * (input_shape.dim(i + 2).dim_value() - 1) +
+            output_padding[i] + kernel_shape[i] - pads[i] -
+            pads[i + n_input_dims];
+        
+
+        final_output_shape->add_dim()->set_dim_value(output_shape_dim);
+      } else if (input_shape.dim(i + 2).has_dim_param()){
+        final_output_shape->add_dim()->set_dim_param(input_shape.dim(i + 2).dim_param());
```

Comment:
Just "final_output_shape->add_dim()" would be correct. We can't set the dim_param to be the same as input_shape.dim(i + 2).dim_param() (say "N") because that implies that these two dimensions are equal, which is not the case.

Reply:
I have resolved this problem.
But I found that seems other ops also have this problem, because on one of my model, I set my input's height and width "*", after the shape inference, the height and width of the output is still "*"while it supposed to be different from the input, so I infer that param means dim_value unknown.